### PR TITLE
CI: Reuse clear cache

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Part of https://github.com/Lightning-AI/utilities/issues/47.

- Removes all existing weekly cache resetting.
- Adds a workflow that clears caches in the repository with one of our reusable workflows, [`clear-cache`](https://github.com/Lightning-AI/utilities/blob/f7303d4a5bc40822f824c41af31bb41d6336eff9/.github/workflows/clear-cache.yml).

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Comment by akihironitta** [comment]
This PR is waiting for a new utilities release that includes `clear-cache` action.
https://github.com/Lightning-AI/utilities/

File: .github/workflows/schedule-clear-cache.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  schedule:
+    # on Sundays
+    - cron: "0 0 * * 0"
+
+jobs:
+  clear-cache:
+    if: ${{ github.repository_owner == 'Lightning-AI' }}
+    uses: Lightning-AI/utilities/.github/workflows/clear-cache.yml@main
+    with:
+      pattern: 'latest|docs'
```
Comment (carmocca) [inline .github/workflows/schedule-clear-cache.yml:13]: What does this do?

**Review by carmocca — APPROVED** [review]
Low confidence approval. How do we know this worked?

File: .github/workflows/schedule-clear-cache.yml (hunk 1 above)
Comment (akihironitta) [reply .github/workflows/schedule-clear-cache.yml:13]: The `'latest|docs'` will be used to grep the cache keys in the repository, and then, the workflow removes caches of grepped keys.

<details><summary>example caches to be cleared weekly</summary>

```bash
$ gh api -H "Accept: application/vnd.github+json" /repos/Lightning-AI/lightning/actions/caches --paginate --jq ".actions_caches[].key" | sort | uniq | grep -E 'latest|docs'
Linux-docs-make-pip-
Linux-docs-test-pip-
Linux-docs-test-pip-tdd19264-
Linux-pip-3.8-latest-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
Linux-pip-3.8-latest-td2752-20d28281bf584b5b194e82c698718338e2c99ac9b6aa28e2ac6cc409b9d4c841
Linux-pip-3.8-latest-td2752-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
Linux-pip-3.8-latest-td2752-c01906eaac36cad9507d97bc62589d8662c57aee933d68311e04bc08866eac15
Linux-pip-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Linux-pip-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Linux-pip-tdd19264-py3.10-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
Linux-pip-tdd19264-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-tdd19264-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Linux-pip-tdd19264-py3.7-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
Linux-pip-tdd19264-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
Linux-pip-tdd19264-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
Windows-pip-3.8-latest-8211468f4f94d870a0890768cc8f1e586df1cfcbcba0f18f56cbdc74a122ffa1
Windows-pip-3.8-latest-td2752-66a7f7498a08d74e669f417bc65392f27f6e444d30edad7afc034c60845d6e25
Windows-pip-3.8-latest-td2752-8211468f4f94d870a0890768cc8f1e586df1cfcbcba0f18f56cbdc74a122ffa1
Windows-pip-3.8-latest-td2752-cb6814fd3b55fdb529db8da122c4525e028a40477352fd78c900f2403c3c0fc1
Windows-pip-py3.10-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-py3.10-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
Windows-pip-py3.7-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-py3.7-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
Windows-pip-td-py3.10-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-td-py3.10-stable-latest-e8976b9aa3ee8564f6c0df352207a35e627ea95ff4c1fb9da86194c45d4501e9
Windows-pip-td-py3.10-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
Windows-pip-td-py3.7-stable-latest-a4f04a36ca240fbb257da322e8f735a978ed8f525cee63bd416bcdb1a238fddc
Windows-pip-td-py3.7-stable-latest-e8976b9aa3ee8564f6c0df352207a35e627ea95ff4c1fb9da86194c45d4501e9
Windows-pip-td-py3.7-stable-latest-ea72290425a2ebd98d47b6e2be209c2a6a1b58d48ef62ccb5bf93d329a3572e3
macOS-pip-3.8-latest-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
macOS-pip-3.8-latest-td2752-20d28281bf584b5b194e82c698718338e2c99ac9b6aa28e2ac6cc409b9d4c841
macOS-pip-3.8-latest-td2752-912b4a7de1f6df12ab94aa2ec56c9614851f7662a0f1d8831f883e89ec11eba6
macOS-pip-3.8-latest-td2752-c01906eaac36cad9507d97bc62589d8662c57aee933d68311e04bc08866eac15
macOS-pip-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
macOS-pip-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
macOS-pip-tdd19264-py3.10-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
macOS-pip-tdd19264-py3.10-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-tdd19264-py3.10-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
macOS-pip-tdd19264-py3.7-stable-latest-6f0f1ad66a753cfc82f88e067cb37e1551d8f51ebb1fd898a7aa9c5f3475784a
macOS-pip-tdd19264-py3.7-stable-latest-c6bdc06d9bc7777942723d8f598bc79660bcd4ceafcf1f91b4a001d120a882da
macOS-pip-tdd19264-py3.7-stable-latest-ec54e63e017d6f3660b72ab3d23d5dc3ebbb72fd42f85b3f14f68ca4ec6b0720
```

</details>

**Review by akihironitta — COMMENTED** [review]
@carmocca Thank you for your review!

> Low confidence approval. How do we know this worked?

We can try running this workflow on this PR once, but that will increase the testing time across (almost) all PRs in this repository. I can test it out this weekend and will paste the link to the job run to make sure that it resets these caches.

**Pull request merged**